archive/imo/imo2008_q3.lean

```diff
@@ -43,7 +43,7 @@ begin
     simp only [int.nat_abs_sq, int.coe_nat_pow, int.coe_nat_succ, int.coe_nat_dvd.mp],
     refine (zmod.int_coe_zmod_eq_zero_iff_dvd (m ^ 2 + 1) p).mp _,
     simp only [int.cast_pow, int.cast_add, int.cast_one, zmod.coe_val_min_abs],
-    rw hy, exact add_left_neg 1 },
+    rw [pow_two, ← hy], exact add_left_neg 1 },
```

Comment:
What happened here? Why was this change necessary?

Reply:
This is because I replaced `∃ y : zmod p, y ^ 2 = -1` in the statement of `zmod.exists_sq_eq_neg_one_iff`with `is_square (-1 : zmod p)`, which unfolds to `∃ y : zmod p, (-1 : zmod p) = y * y`, so we have to convert between `y * y` and `y ^ 2` and also use the symmetric version of the equality.